classes/local/importvideosmanager.php

```diff
@@ -141,15 +147,39 @@ public static function is_enabled_and_working_for_coreimport() {
             return false;
         }
 
-        // Verify that the workflow exists in Opencast.
-        $workflowexists = $apibridge->check_if_workflow_exists($workflow);
+        // Get the import mode.
+        $importmode = get_config('block_opencast', 'importmode');
 
-        // If the workflow does not exist, then the feature is not working.
-        if (!$workflowexists) {
+        if (empty($importmode)) {
             // Inform the caller.
             return false;
         }
 
+        // If Duplicating Events is selected as the import mode.
+        if ($importmode == 'duplication') {
+            // Get the configured duplicate workflow.
+            $workflow = get_config('block_opencast', 'duplicateworkflow');
+    
+            // If the workflow is empty, then the feature is not working.
+            if (empty($workflow)) {
+                // Inform the caller.
+                return false;
+            }
+    
+            // Verify that the workflow exists in Opencast.
+            $workflowexists = $apibridge->check_if_workflow_exists($workflow);
+    
+            // If the workflow does not exist, then the feature is not working.
+            if (!$workflowexists) {
+                // Inform the caller.
+                return false;
+            }
+        } else if ($importmode == 'acl') {
+            // If ACL Change is selected as the import mode.
+
+            // TODO: Add necessary validations for ACL here...
```

Comment:
What about the Todo? Is there missing something here?

Reply:
Currently, there is nothing there, but I wrote it for future implementations. We can remove it if not wanted!